judo-runtime-core-dispatcher/src/main/java/hu/blackbelt/judo/runtime/core/dispatcher/behaviours/ExportCall.java

```diff
@@ -0,0 +1,199 @@
+        queryCustomizerParameterProcessor = new QueryCustomizerParameterProcessor<>(asmUtils, caseInsensitiveLike, identifierProvider, coercer);
+    }
+
+    @Override
+    public boolean isSuitableForOperation(EOperation operation) {
+        return AsmUtils.getBehaviour(operation).filter(o -> o == AsmUtils.OperationBehaviour.EXPORT).isPresent();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object callInRollbackTransaction(final Map<String, Object> exchange, final EOperation operation) {
+
+        CallInterceptorUtil<ListCallPayload<ID>, Object> callInterceptorUtil = new CallInterceptorUtil<>(
+                ListCallPayload.class, Object.class, asmModel, operation, interceptorProvider
+        );
+
+        final boolean bound = AsmUtils.isBound(operation);
+
+        final EReference owner = (EReference) asmUtils.getOwnerOfOperationWithDefaultBehaviour(operation)
+                .orElseThrow(() -> new IllegalArgumentException("Invalid model"));
+
+        final Optional<Map<String, Object>> queryCustomizerParameter = operation.getEParameters().stream().map(ENamedElement::getName)
+                .findFirst()
+                .map(inputParameter -> (Map<String, Object>) exchange.get(inputParameter));
+
+        final DAO.QueryCustomizer<ID> queryCustomizer = queryCustomizerParameterProcessor.build(queryCustomizerParameter.orElse(null), owner.getEReferenceType());
+
+        ListCallPayload<ID> inputParameter = callInterceptorUtil.preCallInterceptors(ListCallPayload.<ID>builder()
+                        .instance(Payload.asPayload(exchange))
+                        .owner(owner)
+                        .queryCustomizer(queryCustomizer)
+                        .build());
+
+        Object result = null;
+
+        if (callInterceptorUtil.shouldCallOriginal()) {
+            if (AsmUtils.annotatedAsTrue(inputParameter.getOwner(), "access")
+                    && inputParameter.getOwner().isDerived()
+                    && asmUtils.isMappedTransferObjectType(inputParameter.getOwner().getEContainingClass())) {
+                checkArgument(!bound, "Operation must be unbound");
+
+                final Map<String, Object> actor;
+                if (exchange.containsKey(Dispatcher.ACTOR_KEY)) {
+                    actor = (Map<String, Object>) exchange.get(Dispatcher.ACTOR_KEY);
+                } else if (exchange.get(Dispatcher.PRINCIPAL_KEY) instanceof JudoPrincipal) {
+                    actor = actorResolver.authenticateByPrincipal((JudoPrincipal) exchange.get(Dispatcher.PRINCIPAL_KEY))
+                            .orElseThrow(() -> new IllegalArgumentException("Unknown actor"));
+                } else {
+                    throw new IllegalStateException("Unknown or unsupported actor");
+                }
+
+                final ID id = (ID) actor.get(identifierProvider.getName());
+
+                try {
+                    result = exporter.exportToInputStream(null,
```

Comment:
Only 1 parameter is different in all 4-5 calls. Maybe we could extract the call, and only collect parameters first, then feed them to the call?

Reply:
[43c8e2e](https://github.com/BlackBeltTechnology/judo-runtime-core/pull/157/commits/43c8e2e290426a21645c58c947e396b4eaace17a)